### ISSUE TITLE
[0.7-ca] missing signalinfo in Webinterface

### ISSUE THREAD

**Issue by 3PO** [catalinii/minisatip#374]
Hi,

After a restart, the signalinfo are shown:

![2017-08-05_081120](https://user-images.githubusercontent.com/945108/28993454-ef18127c-79b6-11e7-9fc7-770b1b175adc.png)

but after a while they are missing:

![2017-08-04_205244](https://user-images.githubusercontent.com/945108/28993461-06321886-79b7-11e7-8b98-1dcfa6f73a17.png)

Log: http://dreipo.cc/logs/minisatip4.log.tgz

BTW: The Channelnames are missing too on most Channels

**Comment by Yuri666** [comment]
Same here on my one minisatip server (on another server all works fine)
After stop showing signal information also stops calculate_bw.

**Comment by Yuri666** [comment]
I'm was wrong. Signal and BW calculations stopped without any visible reason:

[07/08 01:28:47.444 signal]: BW 708KB/s, Total BW: 363 MB, ns/read 1510, r: 19, w: 100 fw: 0, tt: 28 ms
[07/08 01:28:48.447 signal]: BW 659KB/s, Total BW: 364 MB, ns/read 1322, r: 20, w: 97 fw: 0, tt: 26 ms
[07/08 01:28:48.581 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:28:48.582 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:28:49.449 signal]: BW 702KB/s, Total BW: 365 MB, ns/read 1408, r: 20, w: 103 fw: 0, tt: 28 ms
[07/08 01:28:50.451 signal]: BW 722KB/s, Total BW: 365 MB, ns/read 1536, r: 19, w: 102 fw: 0, tt: 29 ms
[07/08 01:28:51.453 signal]: BW 572KB/s, Total BW: 366 MB, ns/read 1166, r: 20, w: 92 fw: 0, tt: 23 ms
[07/08 01:28:52.455 signal]: BW 572KB/s, Total BW: 367 MB, ns/read 1059, r: 20, w: 86 fw: 0, tt: 21 ms
[07/08 01:28:53.458 signal]: BW 640KB/s, Total BW: 367 MB, ns/read 1369, r: 19, w: 98 fw: 0, tt: 26 ms
[07/08 01:28:54.460 signal]: BW 716KB/s, Total BW: 368 MB, ns/read 1428, r: 20, w: 102 fw: 0, tt: 28 ms
[07/08 01:28:55.462 signal]: BW 792KB/s, Total BW: 369 MB, ns/read 1603, r: 20, w: 120 fw: 0, tt: 32 ms
[07/08 01:28:56.464 signal]: BW 1286KB/s, Total BW: 370 MB, ns/read 2407, r: 22, w: 179 fw: 0, tt: 52 ms
[07/08 01:29:03.587 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:29:03.588 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:29:18.626 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:29:18.626 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:29:33.662 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:29:33.663 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:29:48.705 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:29:48.706 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:30:03.742 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):
[07/08 01:30:03.743 main]: reply -> 15 (192.168.1.2:58076) CL:0 [sock_id 9]:
[07/08 01:30:18.754 main]: read RTSP (from handle 15 sock_id 9, len: 0, sid 0):

**Comment by catalinii** [comment]
Hi,

can you check now and provide logs.

Yuri, if the ioctl blocks, there is not much I can do.

Thanks

**Comment by 3PO** [comment]
Hi,
STR seems working now, SNR shows incorrect values.
Channelnames are still missing on most channels:


![2017-08-13_091057](https://user-images.githubusercontent.com/945108/29246945-bc415ca8-8007-11e7-87dd-fb2919a10b58.png)

**Comment by catalinii** [comment]
Hi

Can you upload logs?

-l pmt,dvb

**Comment by 3PO** [comment]
>Can you upload logs?
>
>-l pmt,dvb

--> https://paste.pound-python.org/show/Kb6xEyH28mcFat8tdMdq/

**Comment by catalinii** [comment]
Now I checked closer, and the reason you do not have any channel name is because there is no channel played (none of the pids from the screenshots are Audio or video pids).

The initial issue I saw here was:

[05/08 08:14:26.590 signal]: ioctl FE_READ_BER failed (Operation not supported)
[05/08 08:14:26.591 signal]: ioctl FE_READ_SIGNAL_STRENGTH failed (Operation not supported)
[05/08 08:14:26.591 signal]: ioctl FE_READ_SNR failed (Operation not supported)

Which I wanted to see if it can be reproduced (as initially it was working).

About the signal. Now the new api has higher priority than the old one, and the driver is able to report the strength is reported in dBm (negative value) and the snr as positive value. Basically I add to the strength 100 to be > 0 and then report it. 

In this commit I have added strength_multiplier and snr_multiplier to be able to customize what value is reported. The parameter is -M *:X-Y
Where X is the strength_multiplier, Y, snr_multiplier.

The final value of the strength and the snr will be multipliers and then reported for the DVB adapters.

Thanks
